task_processing/plugins/mesos/retrying_executor.py

```diff
@@ -108,3 +123,38 @@ def stop(self):
 
     def get_event_queue(self):
         return self.dest_queue
+
+    def _get_encoded_task_config(self, task_config):
+        return task_config.set(uuid='{id}-retry{attempt}'.format(
+            id=task_config.uuid,
+            attempt=self.task_retries[task_config.task_id]
+        ))
+
+    def _decode_event(self, e, original_task_id):
+        # Fix task_id references
+        mesos_status = e.raw
+        mesos_status.task_id.value = original_task_id
```

Comment:
If users are using the raw event for something, then they will be a discrepancy between task_id in the event and task_id in status update. I am modifying the raw event to avoid this. But, I don't have a strong opinion about this. Accepted your suggestion for now. If any user complains about this, then we can restore the original task_id.